src/frontend/Ast_to_Mir.ml

```diff
@@ -197,7 +197,7 @@ let param_size transform sizedtype =
   let rec shrink_eigen_mat f st =
     match st with
     | SArray (t, d) -> SArray (shrink_eigen_mat f t, d)
-    | SMatrix (d1, d2) -> SVector (f d1 d2)
+    | SMatrix (d1, d2) | SSparseMatrix (_, _, d1, d2) -> SVector (f d1 d2)
```

Comment:
I believe those two aren't the dimensions, those are the nonzero rows and columns, right?

You definitely don't want to do the same thing for sparse matrix that you do for matrix, even if you had the correct dimension.

Reply:
I just put those there because I have no idea what this function does

Reply:
In the spirit of teaching a person to fish, I'd highly recommend writing some expect tests and playing around with it. I think I showed you those on the call the other day, right? If not, you basically just write something like:
```ocaml
let%expect_test "some name" =
<code that does stuff and then prints it>
```
and then run `dune runtest src` (assuming its' in a `src` file) and it will spit out what it should print. it's like printf based unit testing and I promise it's way more useful than it sounds, haha.

The `shrink_eigen` function (which probably needs a better name) just applies a function to an inner nested eigen type in the case of a matrix transformation like `cholesky_cov`. If we're going with `sparse_matrix` as its own type, then you can't express things like a sparse cholesky_cov matrix so you can just raise an exception here.

Reply:
Here's an example of an expect test

https://github.com/stan-dev/stanc3/blob/eb6893379652852ea0132ddef2c90b00914e3831/src/middle/Middle.ml#L228-L256

Reply:
A better name for this may be `reshape_eigen` with note saying this is used in special functions to reshape special cases of eigen matrices.